app/stores/ada/HWVerifyAddressStore.js

```diff
@@ -0,0 +1,177 @@
+  @observable error: ?LocalizableError = null;
+  @observable selectedAddress: ?{ address: string, path: BIP32Path } = null;
+
+  ledgerBridge: ?LedgerBridge;
+
+  setup() {
+    const actions = this.actions[environment.API].hwVerifyAddress;
+    actions.selectAddress.listen(this._selectAddress);
+    actions.verifyAddress.listen(this._verifyAddress);
+    actions.closeAddressDetailDialog.listen(this._closeAddressDetailDialog);
+  }
+
+  @action _verifyAddress = async (params: { wallet: Wallet }): Promise<void> => {
+    Logger.info('AddressStore::_verifyAddress called');
+
+    if (!this.selectedAddress) {
+      throw new Error('AddressStore::_verifyAddress called with no address selected');
+    }
+    // need to unwrap observable otherwise bridge will fail
+    const path = toJS(this.selectedAddress.path);
+
+    if (!params.wallet.hardwareInfo) {
+      throw new Error('AddressStore::_verifyAddress called with no hardware wallet active');
+    }
+
+    this._setError(null);
+    this._setActionProcessing(true);
+
+    try {
+      // TODO: don't use hardcoded strings and maybe find a better way to do this?
+      if (params.wallet.hardwareInfo.vendor === 'ledger.com') {
+        await this.ledgerVerifyAddress(path);
+      } else if (params.wallet.hardwareInfo.vendor === 'trezor.io') {
+        // not implemented yet
+        await this.trezorVerifyAddress(path);
+      } else {
+        throw new Error('AddressStore::_verifyAddress called with unrecognized hardware wallet');
+      }
+    } catch (error) {
+      Logger.error('AddressStore::_verifyAddress::error: ' + stringifyError(error));
+      this._setError(this._convertToLocalizableError(error));
+    } finally {
+      this._setActionProcessing(false);
+    }
+  }
+
+  trezorVerifyAddress = async (
+    path: BIP32Path
+  ): Promise<void> => {
+    /**
+     * NOT IMPLEMENTED YET
+     */
+    throw new Error('AddressStore::_verifyAddress::Trezor not implemented yet');
+  }
+
+  ledgerVerifyAddress = async (
+    path: BIP32Path,
+  ): Promise<void> => {
+    if (this.ledgerBridge == null) {
+      Logger.info('AddressStore::_verifyAddress new LedgerBridge created');
+      this.ledgerBridge = new LedgerBridge();
+    }
+
+    try {
+      if (this.ledgerBridge) {
+        // trick to fix flow
+        const ledgerBridge: LedgerBridge = this.ledgerBridge;
+
+        await prepareLedgerBridger(ledgerBridge);
+        Logger.info('AddressStore::_verifyAddress show path ' + JSON.stringify(path));
+        await ledgerBridge.showAddress(path);
+      } else {
+        throw new Error(`LedgerBridge Error: LedgerBridge is undefined`);
+      }
+    } finally {
+      Logger.info('HWVerifyStore::ledgerVerifyAddress finalized ');
+    }
+  }
+
+  /**
+   * TODO: This needs to be moved to a single file (also from LedgerSendStore)
+   */
+
+  /** Converts error(from API or Ledger API) to LocalizableError */
+  _convertToLocalizableError = (error: any): LocalizableError => {
```

Comment:
Let's avoid having to copy paste this functio

Reply:
yeah, there is a TODO 3 lines above this so it’s wrapped up before PR review